packages/sdk-node/src/authenticateRequest.ts

```diff
@@ -56,3 +81,26 @@ export const decorateResponseWithObservabilityHeaders = (res: ServerResponse, re
   requestState.reason && res.setHeader(constants.Headers.AuthReason, encodeURIComponent(requestState.reason));
   requestState.status && res.setHeader(constants.Headers.AuthStatus, encodeURIComponent(requestState.status));
 };
+
+const isDevelopmentFromApiKey = (apiKey: string): boolean =>
+  apiKey.startsWith('test_') || apiKey.startsWith('sk_test_');
+
+const getRequestUrl = (req: IncomingMessage): URL => {
+  return new URL(req.url as string, `${getRequestProto(req)}://${req.headers.host}`);
+};
+
+const getRequestProto = (req: IncomingMessage): string => {
```

Comment:
❓ What about [request.protocol](https://nodejs.org/docs/latest-v18.x/api/http.html#requestprotocol)?

Reply:
`request.protocol` is populated but the because `authenticateRequest` accepts request as `IncomingMessage` protocol is not part of the type.

Should we update the type to `http.ClientRequest` ?. 

@dimkl I saw that you updated that type from `ExpressRequest` to  `IncomingMessage`. Any particular reason why we did not go with `http.ClientRequest` back then ?

Reply:
based on the docs the `ExpressRequest` extends the `http.IncomingMessage`, not the `http.ClientRequest` interface.